### PR TITLE
issue in input fields to deposit/withdraw the money

### PR DESCRIPTION
**Pull request by KalyaniPayyavula**
image

image

image

image

image

### CODE REVIEW

File: src/Components/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+export const BackToHomePage = () =>{
+    const navigate = useNavigate()
+
+    const clickHandle = () =>{
+        navigate('/home')
+    }
+    return(
+     <Button variant="contained" onClick={clickHandle}>Back to Home</Button>
+    )
+}
```
Comment (yugandhar-pathi) [inline src/Components/Button/Button.js]: Fix Component Name!! You should not name that as `Button`!! May be `NavigateHomePage` will be better.

File: src/Components/Home/Home.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    padding: "1%",
+}
+    return(
+        <HomeStyle> 
+            <NavLink to='/balance' style={NavStyle}>BALANCE</NavLink>
+            <NavLink to="/deposit" style={NavStyle}>DEPOSIT</NavLink>
+            <NavLink to="/withdraw" style={NavStyle}>WITHDRAW</NavLink>
+       <Button variant="outlined" onClick={LogoutHandle}>Logout</Button>
+       </HomeStyle>
+    )
```
Comment (yugandhar-pathi) [inline src/Components/Home/Home.js]: This is not a component !! It is route. You need to create a folder called `route` and move this under that!!

File: src/Components/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    return(
+        <>
+        <Column>
+         <TextField id="username-id" label="Username" variant="standard" />
+         <TextField id="password-id" label="Password" variant="standard" />
+         <ButtonStyle variant="contained" onClick={loginHandle}>Login</ButtonStyle>
+        </Column>
+       </>
+    )
+}
```
Comment (yugandhar-pathi) [inline src/Components/Login/Login.js]: This is not a component right!! why did you create this under `Components` folder !!

File: src/Components/index.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Login } from "./Login/Login";
+import { Column } from '../Components/Column/Column'
+import { Home } from '../Components/Home/Home'
+import { Deposit } from '../Store/Deposit/Deposit'
+import { Withdraw } from '../Store/Withdrawl/Withdraw'
+import { Balance } from '../Store/Balance/Balance'
+import { BackToHomePage } from '../Components/Button/Button'
+import { Row } from '../Components/Row/Row'
+
+export { Login, Column, Home, Deposit, Withdraw, Balance, BackToHomePage, Row }
```
Comment (yugandhar-pathi) [inline src/Components/index.js]: You can directly `export`.

```
export {Login} from "./Login/Login";
```

File: src/Store/Balance/Balance.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+export const Balance = () =>{
+  const balance = useSelector((state) => state.cash.value)
+    return(
+        <Row>
+          <h2>Your available balance is : {balance}</h2>
+          <BackToHomePage></BackToHomePage>
+        </Row>
+        
+    )
+}
```
Comment (yugandhar-pathi) [inline src/Store/Balance/Balance.js]: Balance is route. Remove this from `store` folder and add it to `routes` folder.

File: src/Store/Deposit/Deposit.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    return(
+        <Row>
+         <TextField label="deposit" value={val}  variant="outlined" onChange={(e) => setVal(e.target.value)}/>
+         <Button variant="contained" onClick={depositHandle}>+</Button>
+         <div>Your available balance is : {balance}</div>
+         <BackToHomePage></BackToHomePage>
+         </Row>
+    )
+}
```
Comment (yugandhar-pathi) [inline src/Store/Deposit/Deposit.js]: Same here!! put this in the right folder.

File: src/Store/Withdrawl/Withdraw.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  }
+    return(
+        <Row>
+          <TextField label="withdrawl" value={val} type="number" variant="outlined" onChange={(e) => setVal(e.target.value)}/>
+          <div>Available balance is : {balance}</div>
+          <Button variant="contained" onClick={withdrawHandle}>-</Button>
+          <BackToHomePage></BackToHomePage>
+        </Row>       
+    )
+}
```
Comment (yugandhar-pathi) [inline src/Store/Withdrawl/Withdraw.js]: Same with this file as well!! Put that in the right folder!!

File: src/Store/userActionSlice.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export const UserActionSlice = createSlice({
+    name : 'cash',
+    initialState : {
+        value : 0
+    },
+
+    reducers : {
+        balance : (state, action) =>{
```
Comment (yugandhar-pathi) [inline src/Store/userActionSlice.js]: Why do you need `balance` action?? You are just reading balance right. I think you only need deposit and withdrawl actions.

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: src/Components/Column/Column.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import PropTypes  from "prop-types"
+import { styled } from "@mui/system"
+
+export const Column = ({children}) =>{
```
Comment (yugandhar-pathi) [inline src/Components/Column/Column.js]: add `dataTestId` ..add for other components as well ..

File: src/Components/NavigateHomePage/NavigateHomePage.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+import { useNavigate } from "react-router-dom"
+
+export const NavigateHomePage = () =>{
+    const navigate = useNavigate()
+
+    const clickHandle = () =>{
+        navigate('/home')
+    }
+    return(
+     <Button variant="contained" onClick={clickHandle}>Back to Home</Button>
```
Comment (yugandhar-pathi) [inline src/Components/NavigateHomePage/NavigateHomePage.js]: Don't hard code string in components!!

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: src/Components/Row/Row.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+        flexDirection: "row",
+        justifyContent: "space-around",
+    })
+    return(
+        <RowStyle data-testid = {dataTestId}>{children}</RowStyle>
+    )
+}
+
+Row.propTypes = {
+    children : PropTypes.element.isRequired
```
Comment (yugandhar-pathi) [inline src/Components/Row/Row.js]: why is `dataTestId` missing in prop types ??

File: src/App.js [hunk 12]
```diff
@@ -1,25 +1,28 @@
-import logo from './logo.svg';
-import './App.css';
+import {Home, Deposit, Withdraw, Balance} from '../src/Components/'
```
Comment (yugandhar-pathi) [inline src/App.js]: Why are you importing pages from `components` ??

File: src/Components/index.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import { Column } from '../Components/Column/Column'
+import { Home } from '../Routes/Home/Home'
+import { Deposit } from '../Routes/Deposit/Deposit'
+import { Withdraw } from '../Routes/Withdraw/Withdraw'
+import { Balance } from '../Routes/Balance/Balance'
+import { NavigateHomePage } from '../Components/NavigateHomePage/NavigateHomePage'
```
Comment (yugandhar-pathi) [inline src/Components/index.js]: Remove pages from here ..they are not components.

File: src/Context/ComponentOne.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+import { createContext } from "react"
+import { ComponentTwo } from "./ComponentTwo"
+
+export const UserContext = createContext()
```
Comment (yugandhar-pathi) [inline src/Context/ComponentOne.js]: Context should be created in a separate file.

You only showed how to read from context, you also need to add how to set to context.

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: src/Components/index.js [hunk 15]
```diff
@@ -0,0 +1,5 @@
+import { Column } from '../Components/Column/Column'
+import { NavigateHomePage } from '../Components/NavigateHomePage/NavigateHomePage'
+import { Row } from '../Components/Row/Row'
+
+export { Column, NavigateHomePage, Row }
```
Comment (yugandhar-pathi) [inline src/Components/index.js]: Why do you need an extra `import` and `export` ??

You can directly export , right ??
```
export { Column } from '../Components/Column/Column'
```

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: src/Context/ComponentOne.js [hunk 16]
```diff
@@ -0,0 +1,15 @@
+import { ComponentTwo } from "./ComponentTwo";
+import { UserContext } from './Context';
+import { useState } from 'react';
+
+export const ComponentOne = () =>{
+    const [val, setVal] = useState("I'm being shared")
```
Comment (yugandhar-pathi) [inline src/Context/ComponentOne.js]: state of the application should not be wrapped inside a component!! It should be moved to Context file.  Check the example I shared.

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 17]
```diff
@@ -1,25 +1,28 @@
-import logo from './logo.svg';
-import './App.css';
+/* eslint-disable react/react-in-jsx-scope */
```
Comment (yugandhar-pathi) [inline src/App.js]: For React 17, you need to add a rule in `.eslintrc.js` .. you need not add this rule in every file.

pls refer .. https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/react-in-jsx-scope.md

File: src/Components/Column/Column.js [hunk 18]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable react/react-in-jsx-scope */
```
Comment (yugandhar-pathi) [inline src/Components/Column/Column.js]: You need to remove this rule from all the files.

File: src/prettier.config.js [hunk 19]
```diff
@@ -0,0 +1,7 @@
+// eslint-disable-next-line no-undef
+module.exports = {
```
Comment (yugandhar-pathi) [inline src/prettier.config.js]: Why did you add multiple configurations for `prettier` ??

File: src/.prettierrc.toml [hunk 20]
```diff
@@ -0,0 +1,4 @@
+trailingComma = "es5"
```
Comment (yugandhar-pathi) [inline src/.prettierrc.toml]: I see multiple configurations for `prettier`..have one file.

File: src/.prettierrc [hunk 21]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (yugandhar-pathi) [inline src/.prettierrc]: I see multiple configurations for prettier. Refer .. `https://prettier.io/docs/en/configuration.html`

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: src/Components/Button/Button.js (hunk 1 above)
Comment (KalyaniPayyavula) [reply src/Components/Button/Button.js]: done

File: src/Components/Home/Home.js (hunk 2 above)
Comment (KalyaniPayyavula) [reply src/Components/Home/Home.js]: done

File: src/Components/Login/Login.js (hunk 3 above)
Comment (KalyaniPayyavula) [reply src/Components/Login/Login.js]: done

File: src/Components/index.js (hunk 4 above)
Comment (KalyaniPayyavula) [reply src/Components/index.js]: done

File: src/Store/Balance/Balance.js (hunk 5 above)
Comment (KalyaniPayyavula) [reply src/Store/Balance/Balance.js]: done

File: src/Store/Deposit/Deposit.js (hunk 6 above)
Comment (KalyaniPayyavula) [reply src/Store/Deposit/Deposit.js]: done

File: src/Store/Withdrawl/Withdraw.js (hunk 7 above)
Comment (KalyaniPayyavula) [reply src/Store/Withdrawl/Withdraw.js]: done

File: src/Store/userActionSlice.js (hunk 8 above)
Comment (KalyaniPayyavula) [reply src/Store/userActionSlice.js]: done

File: src/Components/Column/Column.js (hunk 9 above)
Comment (KalyaniPayyavula) [reply src/Components/Column/Column.js]: done

File: src/Components/NavigateHomePage/NavigateHomePage.js (hunk 10 above)
Comment (KalyaniPayyavula) [reply src/Components/NavigateHomePage/NavigateHomePage.js]: done

File: src/Components/Row/Row.js (hunk 11 above)
Comment (KalyaniPayyavula) [reply src/Components/Row/Row.js]: done

File: src/App.js (hunk 12 above)
Comment (KalyaniPayyavula) [reply src/App.js]: done

File: src/Components/index.js (hunk 13 above)
Comment (KalyaniPayyavula) [reply src/Components/index.js]: done

File: src/Context/ComponentOne.js (hunk 14 above)
Comment (KalyaniPayyavula) [reply src/Context/ComponentOne.js]: done

File: src/Components/index.js (hunk 15 above)
Comment (KalyaniPayyavula) [reply src/Components/index.js]: done

File: src/Context/ComponentOne.js (hunk 16 above)
Comment (KalyaniPayyavula) [reply src/Context/ComponentOne.js]: done

File: src/App.js (hunk 17 above)
Comment (KalyaniPayyavula) [reply src/App.js]: added

File: src/Components/Column/Column.js (hunk 18 above)
Comment (KalyaniPayyavula) [reply src/Components/Column/Column.js]: removed

File: src/prettier.config.js (hunk 19 above)
Comment (KalyaniPayyavula) [reply src/prettier.config.js]: removed

File: src/.prettierrc.toml (hunk 20 above)
Comment (KalyaniPayyavula) [reply src/.prettierrc.toml]: removed

File: src/.prettierrc (hunk 21 above)
Comment (KalyaniPayyavula) [reply src/.prettierrc]: fixed

**Review by yugandhar-pathi — APPROVED** [review]
